Share The Load/Services/PersistanceService.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  PersistanceService.swift
+//  Share The Load
+//
+//  Created by Christopher Hefferman on 13/01/2021.
+//
+
+import Foundation
+import RealmSwift
+
+protocol PersistanceServiceType {
+    var answeredIds: [Int] { get }
+    func save(ids: [Int])
+    func save(questions: [Question])
+    func getQuestions() -> [Question]
+}
+
+/// Massive temporary hack to get save logic working
+/// We'll need a UserService to always give us the same instance of a user rather than this global properties
+/// Global properties are very bad.
+var user = User()
+
+class PersistanceService: PersistanceServiceType {
+    
+    var answeredIds: [Int] = []
+    
+    func save(ids: [Int]) {
+        for id in ids {
+            user.appendAnsweredQuestion(id: id)
+        }
+        
+        do {
+            let realm = try Realm()
+            try realm.write {
+                realm.add(user)
+            }
+        } catch {
+            print("Error: \(error)")
+        }
+    }
+    
+    
+    func save(questions: [Question]) {
+        do {
+            let realm = try Realm()
+            try realm.write {
+                realm.add(questions)
+            }
+            print("Saved to Realm")
+        } catch {
+            print("Error: \(error)")
+        }
+    }
+    
+    
+    func getQuestions() -> [Question] {
+        var questionsFromRealm = [Question()]
```

Comment:
This initialises an array with an empty question as the first element. We ideally want an empty array so you can do var questionsFromRealm: [Question] = []

Reply:
Sounds good - updated.